sdk/tracing/azotel/otel.go

```diff
@@ -0,0 +1,120 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// azotel adapts OpenTelemetry tracing for consumption by the azcore/tracing package.
+package azotel
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/tracing"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	tracesdk "go.opentelemetry.io/otel/sdk/trace"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// TracingProviderOptions contains the optional values for NewTracingProvider.
+type TracingProviderOptions struct {
+	// for future expansion
+}
+
+// NewTracingProvider creates a new tracing.Provider that wraps the specified OpenTelemetry TracerProvider.
+//   - tracerProvider - the TracerProvider to wrap
+//   - opts - optional configuration. pass nil to accept the default values
+func NewTracingProvider(tracerProvider *tracesdk.TracerProvider, opts *TracingProviderOptions) tracing.Provider {
```

Comment:
Small thing - given the similarity in names between our types and theirs perhaps we can alias 'tracesdk' as 'oteltrace' or something?

Reply:
The clash here is due to `go.opentelemetry.io/otel` having two packages named `trace`, `go.opentelemetry.io/otel/sdk/trace` and `go.opentelemetry.io/otel/trace`.  I'm not opposed to changing it, but I chose `tracesdk` as I've seen this in other OTel samples for Go.

Reply:
I see, makes sense.

It looks like you could rename `tracesdk` to `otel` or something since the other `trace` package isn't exported via the public surface here.

Reply:
Good idea.  It could end up colliding with the `otel` package in the future, but we'll deal with it then (if that ever happens).

Reply:
Maybe `otelsdk`.

Reply:
Done.